src/KubernetesClient/KubernetesClientConfiguration.cs

```diff
@@ -76,5 +77,7 @@ public partial class KubernetesClientConfiguration
         /// </summary>
         /// <value>The access token.</value>
         public string AccessToken { get; set; }
```

Comment:
I feel like instead of both having an AccessToken field and a TokenProvider field, we should just switch to TokenProvider everywhere and use a `StaticTokenProvider` or somesuch for the static token case.

Reply:
The issue atm is tokens actually expire. This is very noticeable with GCP provider which issues short lived tokens - KubernetesClient stops working after about 10 minutes. After that token is done, you need to interact with kubectl to force it to get a new token, and re-read it by recreating KubernetesClient.

Reply:
Yeah, I get the problem, what I'm suggesting is that we remove the `AccessToken` field in favor of a `TokenProvider` everywhere.

Reply:
I'm trying to do this for AKS, but the cluster that I just spun up seems to use certs for authentication, as I got client-key-data, client-certificate-data, and token in my config file. The current code seems to be written where there's access_token. Under what scenario would this be so?

I feel like if we're supporting kubectl current context authentication, then we need to support all different types of authentication that kubectl uses, which includes
- OIDC - discover endpoints automatically
- external CLI to get token (like gcp)
- hardcoded / static (aks seems to use static token + cert)
- Oauth2 - hardcoded endpoints

Would be good to see a sample of what AWS EKS config looks like, I don't have one.

Reply:
Seems like this is where all the different built in authenticators are defined
https://github.com/kubernetes/client-go/tree/master/plugin/pkg/client/auth

Reply:
Based on my research, it seems there are 3 out of the box authentication providers that kubectl uses: gcp, azure & oidc, and each of these would translate to their own unique token provider. They are as listed in the above URL. These act very much like HttpMessageHandlers. The selection of the provider is dependent on 
```
users:
  user:
    auth-provider:
      name: XXX
```

My implementation for gcp is actually correct. Ideally, we want to have separate implementation for the other two types of providers. I propose that is done in a separate PR.

I won't be able to code for azure plugin because I'm lacking an AKS cluster that is integrated with Azure AD. I can't set this up as you need to be Azure tenant admin to configure this, and I don't have those rights in my org. If someone wants to provide me with AKS cluster that is enabled for Azure AD integration, I would be happy to create that token authenticator. The instructions on how to set this up are here: https://docs.microsoft.com/en-us/azure/aks/azure-ad-integration